### PR TITLE
DATACOUCH-545 - return null if not found instead of throwing an excep…

### PR DESCRIPTION
**Pull request by mikereiche**
…tion

### CODE REVIEW

File: src/main/java/org/springframework/data/couchbase/core/ExecutableFindByIdOperationSupport.java [hunk 1]
```diff
@@ -53,7 +54,12 @@
 
 		@Override
 		public T one(final String id) {
-			return reactiveSupport.one(id).block();
+			try {
+				return reactiveSupport.one(id).block();
+			} catch(DocumentNotFoundException dnfe){
+				dnfe.printStackTrace();
```
Comment (daschl) [inline src/main/java/org/springframework/data/couchbase/core/ExecutableFindByIdOperationSupport.java:60]: this is a leftover I assume?

File: src/main/java/org/springframework/data/couchbase/core/ReactiveFindByQueryOperationSupport.java [hunk 2]
```diff
@@ -70,7 +70,7 @@ public ReactiveFindByQueryOperationSupport(final ReactiveCouchbaseTemplate templ
 
 		@Override
 		public Mono<T> one() {
-			return all().single();
+			return all().singleOrEmpty();
```
Comment (daschl) [inline src/main/java/org/springframework/data/couchbase/core/ReactiveFindByQueryOperationSupport.java:73]: If this is changed here, does it also need to be changed for analytics?

Comment (mikereiche) [reply src/main/java/org/springframework/data/couchbase/core/ReactiveFindByQueryOperationSupport.java:73]: Yes, it probably needs to be changed there as well. I'll take a look.  I'll look for 'single()' everywhere.

File: src/main/java/org/springframework/data/couchbase/core/ExecutableFindByIdOperationSupport.java (hunk 1 above)
Comment (mikereiche) [reply src/main/java/org/springframework/data/couchbase/core/ExecutableFindByIdOperationSupport.java:60]: I think it needs to be here in addition to ReactiveFindByQueryOperationSupport.  I had myself convinced there are different code-paths.  I'll check again.

Comment (daschl) [reply src/main/java/org/springframework/data/couchbase/core/ExecutableFindByIdOperationSupport.java:60]: Sorry, I meant the printStackTrace will dump into the logs and seems to be a leftover of debugging

**Pull request closed without merging**